### PR TITLE
Introduce equipment ranking

### PR DESCRIPTION
**Pull request by area363**
Needs some code adjustment after merging https://github.com/planetarium/NineChronicles.DataProvider/pull/42.

image

### CODE REVIEW

**Comment by area363** [comment]
defaultequipranking
itemsubtype
Updated equipment ranking query.

File: NineChronicles.DataProvider/Store/MySqlStore.cs [hunk 1]
```diff
@@ -341,5 +342,73 @@ public void DeleteItemEnhancement(Guid id)
...
+                                "FROM `Equipments` AS `h` GROUP BY `h`.`AvatarAddress` ");
+            }
+
+            if (limit is { } limitNotNull)
+            {
+                query = query.Take(limitNotNull);
+            }
+
+            var list = query.ToList();
+            return list;
```
Comment (longfin) [inline NineChronicles.DataProvider/Store/MySqlStore.cs:411]:
```suggestion
            return query.ToList();
```

**Review by longfin — APPROVED** [review]

File: NineChronicles.DataProvider/RenderSubscriber.cs [hunk 2]
```diff
@@ -147,15 +223,112 @@ protected override Task ExecuteAsync(CancellationToken stoppingToken)
...
 
                         if (ev.Action is ItemEnhancement itemEnhancement)
                         {
                             MySqlStore.DeleteItemEnhancement(itemEnhancement.Id);
                             Log.Debug("Deleted ItemEnhancement action in block #{index}", ev.BlockIndex);
+                            string avatarName = ev.OutputStates.GetAvatarState(itemEnhancement.avatarAddress).name;
+                            var inventory = ev.OutputStates.GetAvatarState(itemEnhancement.avatarAddress).inventory;
+                            if (inventory is { } inventoryNotNull)
+                            {
+                                ProcessEquipments(
```
Comment (longfin) [inline NineChronicles.DataProvider/RenderSubscriber.cs:251]: IMO, it would be better if we specify itemId for this action and migrate that item only.

**Comment by area363** [comment]
PTAL! @longin :pray:

**Review by longfin — APPROVED** [review]

**Pull request merged**